src/main/java/core/basesyntax/Main.java

```
@@ -1,9 +1,63 @@
+import core.basesyntax.service.impl.ReportGeneratorImpl;
+import core.basesyntax.service.impl.ShopServiceImpl;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.BalanceOperationHandler;
+import core.basesyntax.strategy.impl.OperationStrategyImpl;
+import core.basesyntax.strategy.impl.PurchaseOperationHandler;
+import core.basesyntax.strategy.impl.ReturnOperationHandler;
+import core.basesyntax.strategy.impl.SupplyOperationHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 /**
  * Feel free to remove this class and create your own.
  */
 public class HelloWorld {
-    // HINT: In the `public static void main(String[] args)` it is better to create instances of your classes, 
-    // and call their methods, but do not write any business logic in the `main` method!
+    public static void main(String[] args) {
+        //read data from file
+        FileReaderService fileReaderService = new FileReaderServiceImpl();
+        List<String> fileContent = fileReaderService.readFile("dailyActivities.csv");
+
+        //create StorageDao to pass it to all handlers
+        StorageDao storageDao = new StorageDaoImpl();
+
+        Map<FruitTransaction.Operation, OperationHandler> handlers = new HashMap<>();
+        handlers.put(FruitTransaction.Operation.BALANCE, new BalanceOperationHandler(storageDao));
+        handlers.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler(storageDao));
+        handlers.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperationHandler(storageDao));
+        handlers.put(FruitTransaction.Operation.RETURN, new ReturnOperationHandler(storageDao));
+
+        //get FruitTransaction List from Strings List
+        DataConverter dataConverter = new DataConverterImpl();
+        List<FruitTransaction> transactions =
+                dataConverter.convertStringsToTransactionsList(fileContent);
```

DataConverter is used here to convert file lines to transactions, but DataConverterImpl does not validate the CSV header and does not reject empty fruit names explicitly. The checklist requires to "Skip or throw on invalid header or malformed lines (e.g., wrong number of fields, non-integer quantity, empty fruit name, unknown operation code); must not silently ignore without validation." Update DataConverterImpl so it validates the header string equals "type,fruit,quantity", rejects empty fruit names, negative quantities, and provides informative RuntimeException messages including line numbers or content. See current DataConverterImpl behavior and checklist requirement  .